db/db_impl/db_impl.cc

```diff
@@ -1922,6 +1941,19 @@ void DBImpl::MultiGetImpl(
   PERF_TIMER_STOP(get_post_process_time);
 }
 
+void DBImpl::TriggerReadCompactionIfRequired(
+    ColumnFamilyData* cfd,
+    VersionStorageInfo* storage_info) {
+  assert(cfd != nullptr);
+  assert(storage_info != nullptr);
+  if (storage_info->ShouldTriggerReadCompaction()) {
+    InstrumentedMutexLock l(&mutex_);
```

Comment:
Wondering what would be the impact of this lock here. I guess the benchmarks would tell us. 
(I need to reason through if its possible to get rid of this lock).

Reply:
This should not happen too often and I think there is already a case where DB mutex can get acquired in the read path (when creating super version). But I don't know the frequency of that one. A way to get around taking the mutex here would be to have a separate thread for scheduling read-triggered compactions and only schedule the scheduling here.